modules/openy_gc_auth/modules/openy_gc_auth_custom/src/Plugin/rest/resource/GatedContentCustomAuthentication.php

```diff
@@ -123,19 +139,9 @@ public function post(Request $request) {
     $provider_config = $this->configFactory->get('openy_gc_auth.provider.custom');
     if ($provider_config->get('enable_recaptcha')) {
       // Validate recaptchaToken if enabled in the provider config.
-      if (!$content['recaptchaToken']) {
-        return $this->errorResponse($this->t('ReCaptcha token required.'), 400);
-      }
-
-      $config = $this->configFactory->get('recaptcha.settings');
-      $recaptcha_secret_key = $config->get('secret_key');
-      $recaptcha = new ReCaptcha($recaptcha_secret_key, new Drupal8Post());
-      if ($config->get('verify_hostname')) {
-        $recaptcha->setExpectedHostname($_SERVER['SERVER_NAME']);
-      }
-      $resp = $recaptcha->verify($content['recaptchaToken'], $request->getClientIp());
-      if (!$resp->isSuccess()) {
-        return $this->errorResponse($this->t('ReCaptcha token invalid.'), 400);
+      $validation_result = $this->validateRecaptcha($content, $request);
+      if ($validation_result instanceof ModifiedResourceResponse) {
```

Comment:
what happens if not?
Any error handling in UI?

Reply:
**validation_result** is instanceof ModifiedResourceResponse only in case of error, so we return error response here.